### PR TITLE
coverity maria repair parallel

### PR DESCRIPTION
**Pull request by grooverdan**
I submit this under the MCA.

### CODE REVIEW

**Comment by svoj** [comment]
And then condition must change to ```if (my_b_inited()) end_io_cache();```

**Comment by svoj** [comment]
@grooverdan we have 5.5 week now. Please update if you want to get this into next 5.5 release.

**Comment by grooverdan** [comment]
thanks for the heads up.

File: storage/maria/ma_check.c [hunk 1]
```diff
@@ -4250,6 +4250,8 @@ int maria_repair_parallel(HA_CHECK *param, register MARIA_HA *info,
     }
   */
   DBUG_PRINT("info", ("is quick repair: %d", (int) rep_quick));
+  if (!rep_quick)
+    my_b_clear(&new_data_cache);
```
Comment (svoj) [inline storage/maria/ma_check.c]: Taking into account simplicity of ```my_b_clear()``` and ```my_b_inited()``` I'd rather do ```my_b_clear()``` unconditionally. It saves 2 conditional jumps, which performance wise are worse than single unconditional store.

This is just to voice my opinion, current implementation is generally fine too. No action needed.

**Comment by svoj** [comment]
Thanks!

**Comment by grooverdan** [comment]
works for me. Thanks.

**Pull request merged**